src/main/java/Task.java

```diff
@@ -0,0 +1,58 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Task {
+    protected boolean done;
+    protected String task;
+    public Task (String in) {
+        task = in;
+        done = false;
+    }
+    public void complete() {
+        done = true;
+    }
+    public String saveText() {
+        return done + " | " + task;
+    }
+
+    /**
+     * Converts a line of text in save file to a Task object.
+     * @param in The string to be converted.
+     * @return Task represented by the input string.
+     */
+    public static Task readText(String in) {
+        String[] result = in.split(" \\| ");
+        String taskType = result[0];
+        boolean temp = Boolean.parseBoolean(result[1]);
+        Task tempTask;
+        if (taskType.equals("T")) {
+            tempTask = new ToDo(result[2]);
+        } else if (taskType.equals("D")) {
+            tempTask = new Deadline(result[2],LocalDateTime.parse(result[3],DateTimeFormatter.ofPattern("dd MMM yyyy HHmm")));
```

Comment:
Parameters look quite long and there's many magic numbers, maybe split up the lines and use constants?